hyperspy/misc/eds/utils.py

```diff
@@ -13,11 +13,44 @@ def _get_element_and_line(Xray_line):
 
 
 def _get_energy_xray_line(xray_line):
-    energy, line = _get_element_and_line(xray_line)
-    return elements_db[energy]['Atomic_properties']['Xray_lines'][
+    element, line = _get_element_and_line(xray_line)
+    return elements_db[element]['Atomic_properties']['Xray_lines'][
         line]['energy (keV)']
 
 
+def get_xray_lines_near_energy(energy, spread=0.1):
+    """Find xray lines near a specific energy, more specifically search in
+    range 'energy' +/- 'spread'
+
+    Parameters
+    ----------
+    energy : float
+        Energy to search near in keV
+    spread : float
+        Spread in keV around energy in which to search
+
+    Returns
+    -------
+    List of xray-lines sorted by energy difference to given energy.
+    """
+    valid_lines = []
+    E_min, E_max = energy - spread, energy + spread
+    for element, el_props in elements_db.iteritems():
+        # Not all elements in the DB have the keys, so catch KeyErrors
+        try:
+            lines = el_props['Atomic_properties']['Xray_lines']
+        except KeyError:
+            continue
```

Comment:
The database has been so far build for EELS and EDS purpose. Elements always have atomic properties. Anyway, minor issue. Your version is more stable